src/adlist.c

```diff
@@ -187,6 +189,7 @@ void listDelNode(list *list, listNode *node) {
  * Remove the specified node from the list without freeing it.
  */
 void listUnlinkNode(list *list, listNode *node) {
+    serverAssert(list->len > 0);
```

Comment:
It's just `assert` when we use `serverassert.h`.

```suggestion
    assert(list->len > 0);
```